cpp/examples/arrow/CMakeLists.txt

```diff
@@ -123,6 +123,19 @@ if(ARROW_PARQUET AND ARROW_DATASET)
                     ${DATASET_EXAMPLES_LINK_LIBS})
   add_dependencies(dataset-parquet-scan-example parquet)
 
+  if (ARROW_SKYHOOK)
+    if (ARROW_BUILD_SHARED)
+      set(DATASET_EXAMPLES_LINK_LIBS ${DATASET_EXAMPLES_LINK_LIBS}
```

Comment:
two nits: since we're reusing the same variable, maybe move this to the bottom of the enclosing if block so we don't link the other examples against Skyhook; and use `list(APPEND ...)`?